troupe.go

```diff
@@ -58,73 +61,106 @@ func NewTroupe(cfg Config) (*Troupe, error) {
 		if b, err = NewActor(bCfg); err != nil {
 			return nil, err
 		}
+		b.ID = len(Actors)
 		Actors = append(Actors, b)
 	}
 	return &Troupe{
 		Actors:             Actors,
 		minActors:          cfg.Min,
 		maxActors:          cfg.Max,
 		defaultActorConfig: bCfg,
+		r:                  rand.New(rand.NewSource(time.Now().UnixNano())),
```

Comment:
I had been, but removed it from the branch before I pushed the changes up (see: the ID comment). It's easy to add back in once I have time to keep testing it, but wanted to get some better code up for review sooner than later.